### ISSUE TITLE
Proxmox builder: Crash and docs

### ISSUE THREAD

**Issue by carlpett** [hashicorp/packer#7540]
Created from comment by @zaenk in #7391:

* The [docs](https://github.com/hashicorp/packer/pull/7391/files?file-filters%5B%5D=.md#diff-4d8ad6a99d84244b1840b1c869d3edd4R74) says, possible values of `os` might be `linux`, `windows`, `solaris` or `other`, but setting `linux` (for ProxMox 5.4-3) gives the following >error

```
==> proxmox: Creating VM
==> proxmox: Error creating VM: 400 Parameter verification failed., error status: {"errors":{"ostype":"value 'linux' does not have a value in the enumeration 'other, wxp, w2k, w2k3, w2k8, wvista, win7, win8, win10, l24, l26, solaris'"},"data":null} (params: map[agent:1 cores:1 cpu:host description:Packer ephemeral build VM ide2:local:iso/ubuntu-18.04.2-server-amd64.iso,media=cdrom memory:512 name:base-packer net0:virtio=1A:C6:6E:4A:4B:D0,bridge=vmbr1 onboot:false ostype:linux scsi0:media=disk,size=5G,file=datassd:vm-199-disk-0 sockets:1 >vmid:199])
```

Setting the option to `l26` fixed this issue.

Also the the VM disk image was not cleaned up after this failure.

* With a correct config the build always fails right after boot.

```
proxmox output will be in this color.

==> proxmox: Creating VM
==> proxmox: Starting VM
==> proxmox: Starting HTTP server on port 8986
==> proxmox: Waiting 10s for boot
==> proxmox: Stopping VM
==> proxmox: Deleting VM
Build 'proxmox' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> proxmox: unexpected EOF

==> Builds finished but no artifacts were created.
```

The [crash log](https://gist.github.com/zaenk/790087fc2fd2d6a0c8d004efd282effa) gives away the culprit https://github.com/hashicorp/packer/>blob/master/builder/proxmox/step_type_boot_command.go#L69

`uint` should be used here like [other places in the code](https://github.com/hashicorp/packer/>search?q=state.Get%28%22http_port%22%29&unscoped_q=state.Get%28%22http_port%22%29)

**Comment by carlpett** [comment]
Regarding docs: This seems to be a case of copy/paste error. I'll fix that. The disk not being removed is a bit tricky though, this is on Proxmox's side I believe. We don't get a reference back when the VM fails to be created, so there is no easy way to remove it.

Regarding the uint/int, it seems this was changed between the time I started working on the builder vs when it was merged: #7423. I'll bring the builder into compliance with that too!
